### PR TITLE
BED-4384: Bump Postgres to 16 in development

### PR DESCRIPTION
**Pull request by ALCooper12**
## Description

- Added new just command for only spinning up the database and pgadmin containers 

## Motivation and Context

- Ticket: BED-4384

*Why is this change required? What problem does it solve?*
- This solves the issue of proper data transfer and bumping Postgres from version 13.2 to version 16

## How Has This Been Tested?

- Local testing where running the instruction commands below in my terminal resulted in no data loss being seen in pgadmin/UI, and the bhce docker pod spins up with the proper Postgres 16 image without seeing data directory error logs

### Running from BHCE repo:
```
# 1. First, stop any running containers and start only the database and pgadmin containers
just pg-only

# 2. Change the password encryption from MD5 to SCRAM-SHA-256 format (Postgres 16 defaults to this format)
just pg-only exec app-db sh -c "\"psql -c \\\"SET password_encryption = 'scram-sha-256'; ALTER ROLE CURRENT_USER PASSWORD '\${BH_POSTGRES_PASSWORD:-bloodhoundcommunityedition}';\\\" \""

# 3. Export existing data via a database dump (while the database and pgadmin containers are running)
just pg-only exec app-db pg_dumpall > dump.sql

# 4. Stop with Ctrl+C and remove all containers
just bh-dev down

# 5. Remove only the database volume for the Postgres service
just pg-only down -v

# 6. Edit the docker-compose file from 13.2 to 16 in /docker-compose.dev.yml 

# 7. Start only the database and pgAdmin containers
just pg-only

# 8. Restore the database from the dump file
just pg-only exec -T app-db psql < dump.sql

# 9. Stop the database and pgAdmin containers with Ctrl+C

# 10. Finally, start all containers
just bh-dev
```
### Otherwise:
```
# 1. First, stop any running containers, navigate to examples/docker-compose/, and start only the database container
docker compose up app-db

# 2. Change the password encryption from MD5 to SCRAM-SHA-256 format (Postgres 16 defaults to this format)
docker compose exec app-db sh -c "psql -c \"SET password_encryption = 'scram-sha-256'; ALTER ROLE CURRENT_USER PASSWORD '\${BH_POSTGRES_PASSWORD:-bloodhoundcommunityedition}';\""

# 3. Export existing data via a database dump (while the database container is running)
docker compose exec app-db pg_dumpall > dump.sql

# 4. Stop with Ctrl+C and remove the database container
docker compose rm -f app-db

# 5. Remove only the database volume for the Postgres service
docker volume rm docker-compose_postgres-data

# 6. Edit the docker-compose file from 13.2 to 16 in examples/docker-compose/docker-compose.yml or whatever directory the file is kept

# 7. Start only the database container
docker compose up app-db

# 8. Restore the database from the dump file
docker compose exec -T app-db psql < dump.sql

# 9. Stop the database container with Ctrl+C

# 10. Finally, start all containers
docker compose up
```

## Types of changes

- Chore (a change that does not modify the application functionality)
- Database Migrations

## Checklist:

- [x] I have met the contributing prerequisites
  - Assigned myself to this PR
  - Added the appropriate labels
  - Associated an issue: https://github.com/SpecterOps/BloodHound/issues/672
  - Read the Contributing guide: https://github.com/SpecterOps/BloodHound/wiki/Contributing
- [x] I have ensured that related documentation is up-to-date
  - Open API docs
  - Code comments (GoDocs / JSDocs)
- [x] I have followed proper test practices
  - Added/updated tests to cover my changes
  - All new and existing tests passed

### CODE REVIEW

**Review by computator — APPROVED** [review]

**Review by juggernot325 — APPROVED** [review]

File: docker-compose.dev.yml [hunk 1]
```diff
@@ -28,8 +28,15 @@ services:
       - /var/run/docker.sock:/var/run/docker.sock:ro
 
   app-db:
+    profiles:
+      - dev
+      - api-only
+      - ui-only
+      - debug-api
+      - pg-only
     image: docker.io/library/postgres:13.2
```
Comment (ddlees) [inline docker-compose.dev.yml:37]: Go ahead and update the version to 16

File: examples/docker-compose/docker-compose.yml [hunk 2]
```diff
@@ -18,6 +18,7 @@ services:
   app-db:
     image: docker.io/library/postgres:13.2
```
Comment (ddlees) [inline examples/docker-compose/docker-compose.yml:19]: Go ahead and update the version to 16

File: docker-compose.dev.yml [hunk 3]
```diff
@@ -28,8 +28,15 @@ services:
...
 
   app-db:
-    image: docker.io/library/postgres:13.2
+    profiles:
+      - dev
+      - api-only
+      - ui-only
+      - debug-api
+      - pg-only
+    image: docker.io/library/postgres:16
```
Comment (computator) [inline docker-compose.dev.yml:37]: Don't forget to update this in the `docker-compose.test.yml` as well if we decide to go with the version upgrade

Comment (ALCooper12) [reply docker-compose.dev.yml:37]: Just made the changes!

**Review by ddlees — APPROVED** [review]

**Pull request merged**